rclcpp/src/rclcpp/experimental/executors/events_executor/events_executor.cpp

```diff
@@ -511,9 +511,19 @@ EventsExecutor::add_notify_waitable_to_collection(
 {
   // The notify waitable is not associated to any group, so use an invalid one
   rclcpp::CallbackGroup::WeakPtr weak_group_ptr;
-  collection.insert(
+  bool inserted = collection.insert(
   {
     this->notify_waitable_.get(),
     {this->notify_waitable_, weak_group_ptr}
   });
+  // Explicitly ignore if the notify waitable was not inserted because that means
+  // it was already inserted, which happens initially as it is explicitly added
+  // in the constructor as well as every time the collection is reset, so on
+  // the first reset there is a second insertion attempt.
+  // We could check before trying to insert, but that would require a "find" call
+  // on each refresh, which is expensive, and otherwise it would require additional
+  // state in this class to detect the initial case where it is added twice.
+  // Therefore we just insert and ignore it if it fails (the only way it fails
+  // is when a duplicate is inserted).
+  RCUTILS_UNUSED(inserted);
```

Comment:
This is an oddity I found with the events executor, where it was always inserting the notify waitable twice on one occasion (first refresh after construction), and it wasn't checking the return value of insert, which if we had checked would have occasionally been false meaning it wasn't inserted. This didn't end up mattering, but it did feel weird and broke some things when I added the vector storage along side the map.

Anyway I landed on this approach where we continue to add it twice sometimes, but we just ignore it when it wasn't inserted as the fact that it has been inserted is important, not the order in which it was, at least in the specific case of this notify waitable.

Reply:
FYI @mauropasse

Reply:
I think that this was a bug.
I opened a PR to fix it https://github.com/ros2/rclcpp/pull/2564

Reply:
Cool, thanks for checking up on it. I guess I'll wait for that pr to merge and then update this. I have to check into timer execution order on Windows for these new tests anyways.